frontend/src/routes/Infrastructure/InfraEnvironments/Details/InfraEnvironmentDetailsPage.tsx

```diff
@@ -140,12 +158,60 @@ const InfraEnvironmentDetailsPage: React.FC<InfraEnvironmentDetailsPageProps> =
                             </AcmSecondaryNav>
                         }
                         actions={
-                            <AcmButton
-                                isDisabled={!infraEnv?.status?.isoDownloadURL}
-                                onClick={() => setISOModalOpen(true)}
-                            >
-                                {t('Add host')}
-                            </AcmButton>
+                            <Dropdown
+                                id={`${infraEnv.metadata.name}-actions`}
+                                toggle={
+                                    <DropdownToggle id="dropdown-basic" onToggle={onToggle} isPrimary>
+                                        {t('Add host')}
+                                    </DropdownToggle>
+                                }
+                                isOpen={isKebabOpen}
+                                isText
+                                dropdownItems={[
+                                    <DropdownItem
+                                        key="discovery-iso"
+                                        onClick={() => {
+                                            setIsKebabOpen(false)
+                                            setISOModalOpen(true)
+                                        }}
+                                        description={t('Discover multiple hosts with a single file')}
```

Comment:
The DropdownItem descriptions need to be revisited.
```suggestion
                                        description={t('Discover hosts by booting a discovery image')}
```